🌐 [i18n-KO] Translated `main_classes/model.md` to Korean (#33606)
huggingface/transformers: docs/source/ko/main_classes/model.md

<!--Copyright 2020 The HuggingFace Team. All rights reserved.

Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
the License. You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
specific language governing permissions and limitations under the License.

⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
rendered properly in your Markdown viewer.

-->

# 모델

> 기본 클래스 [`PreTrainedModel`], [`TFPreTrainedModel`], [`FlaxPreTrainedModel`]는 로컬 파일과 디렉토리로부터 모델을 로드하고 저장하거나 또는 (허깅페이스 AWS S3 리포지토리로부터 다운로드된) 라이브러리에서 제공하는 사전훈련 모델 설정값을 로드하고 저장하는 것을 지원하는 기본 메소드를 구현하였습니다.   

Reviewer: ```suggestion
기본 클래스 [`PreTrainedModel`], [`TFPreTrainedModel`], [`FlaxPreTrainedModel`]는 로컬 파일과 디렉토리로부터 모델을 로드하고 저장하거나 또는 (허깅페이스 AWS S3 리포지토리로부터 다운로드된) 라이브러리에서 제공하는 사전 훈련된 모델 설정을 로드하고 저장하는 것을 지원하는 기본 메소드를 구현하였습니다.   
```

> 기본 클래스 [`PreTrainedModel`], [`TFPreTrainedModel`], [`FlaxPreTrainedModel`]는 로컬 파일과 디렉토리로부터 모델을 로드하고 저장하거나 또는 (허깅페이스 AWS S3 리포지토리로부터 다운로드된) 라이브러리에서 제공하는 사전훈련 모델 설정값을 로드하고 저장하는 것을 지원하는 기본 메소드를 구현하였습니다.   

Reviewer 2: ```suggestion
기본 클래스 [`PreTrainedModel`], [`TFPreTrainedModel`], [`FlaxPreTrainedModel`]는 로컬 파일 및 디렉토리에서 모델을 로드하고 저장하거나,
 또는 허깅페이스 AWS S3 리포지토리에서 다운로드한 라이브러리에서 제공하는 사전훈련 모델 설정값으로부터 모델을 로드하고 저장하는 기본 메소드를 구현하였습니다.   
```
가독성을 높이려 표현을 바꿔보았습니다 !

> [`PreTrainedModel`]과 [`TFPreTrainedModel`]은 또한 모든 모델들에 공통적으로 지원하는 몇가지 메소드를 구현하였습니다:

Reviewer 2: ```suggestion
[`PreTrainedModel`]과 [`TFPreTrainedModel`]은 모든 모델들에 공통적으로 지원되는 몇 메소드를 구현하였습니다:
```

> [`PreTrainedModel`]과 [`TFPreTrainedModel`]은 또한 모든 모델들에 공통적으로 지원하는 몇가지 메소드를 구현하였습니다:

Author: ```suggestion
[`PreTrainedModel`]과 [`TFPreTrainedModel`]은 또한 모든 모델들을 공통적으로 지원하는 메소드 여러개를 구현하였습니다:
```

- 새 토큰이 단어장에 추가될 때, 입력 토큰 임베딩의 크기를 조정합니다.

> - 모델의 어텐션 헤드를 가지치기합니다.

Reviewer: ```suggestion
- 모델의 어텐션 헤드를 가지치기 합니다.
```

> 각 모델에 공통인 다른 메소드들은 다음과 같은 클래스에 정의 되어있습니다. 

Reviewer 3: 표현을 바꾸었습니다.

- [`~modeling_utils.ModuleUtilsMixin`](파이토치 모델용)
- 텍스트 생성을 위한 [`~modeling_tf_utils.TFModuleUtilsMixin`](텐서플로 모델용)
- [`~generation.GenerationMixin`](파이토치 모델용)
- [`~generation.FlaxGenerationMixin`](Flax/JAX 모델용)

## PreTrainedModel

[[autodoc]] PreTrainedModel
    - push_to_hub
    - all

> 사용자 정의 모델은 초고속 초기화가 특정 모델에 적용될 수 있는지 여부를 결정하는 `_supports_assign_param_buffer`도 포함해야 합니다.

Author: ```suggestion
사용자 정의 모델은 초고속 초기화(superfast init)가 특정 모델에 적용될 수 있는지 여부를 결정하는 `_supports_assign_param_buffer`도 포함해야 합니다.
```

> 사용자정의 모델은 superfast init이 특정모델에 적용될 수 있는지 여부를 결정하는 `_supports_assign_param_buffer`도 포함해야합니다.

Reviewer: ```suggestion
사용자 정의 모델은 초고속 초기화가 특정 모델에 적용될 수 있는지 여부를 결정하는 `_supports_assign_param_buffer`도 포함해야 합니다.
```
superfast init도 번역을 하면 좋을 것 같아 제안드립니다!😊

> `test_save_and_load_from_pretrained`이 실패하면, 당신의 모델이 `_supports_assign_param_buffer`를 필요로 하는지 확인하세요.

Reviewer 2: ```suggestion
`test_save_and_load_from_pretrained` 실패 시, 모델이 `_supports_assign_param_buffer`를 필요로 하는지 확인하세요.
```
